lib/liquid/context.rb

```diff
@@ -27,7 +27,7 @@ def initialize(environments = {}, outer_scope = {}, registers = {}, rethrow_erro
       @this_stack_used = false
 
       if rethrow_errors
-        self.exception_handler = ->(e) { raise }
+        self.exception_handler = ->(e) { raise e }
```

Comment:
Yep, it fails without `e`:
```
  1) Failure:
TemplateTest#test_undefined_filters_raise [/Users/Jeiwan/Work/liquid/liquid/test/integration/template_test.rb:296]:
[Liquid::UndefinedFilter] exception expected, not
Class: <RuntimeError>
Message: <"">
---Backtrace---
/Users/Jeiwan/Work/liquid/liquid/lib/liquid/context.rb:30:in `block in initialize'
/Users/Jeiwan/Work/liquid/liquid/lib/liquid/context.rb:84:in `handle_error'
/Users/Jeiwan/Work/liquid/liquid/lib/liquid/strainer.rb:52:in `invoke'
/Users/Jeiwan/Work/liquid/liquid/lib/liquid/context.rb:103:in `invoke'
/Users/Jeiwan/Work/liquid/liquid/lib/liquid/variable.rb:78:in `block in render'
/Users/Jeiwan/Work/liquid/liquid/lib/liquid/variable.rb:76:in `each'
/Users/Jeiwan/Work/liquid/liquid/lib/liquid/variable.rb:76:in `inject'
/Users/Jeiwan/Work/liquid/liquid/lib/liquid/variable.rb:76:in `render'
/Users/Jeiwan/Work/liquid/liquid/lib/liquid/block_body.rb:93:in `render_node'
/Users/Jeiwan/Work/liquid/liquid/lib/liquid/profiler/hooks.rb:5:in `block in render_node_with_profiling'
/Users/Jeiwan/Work/liquid/liquid/lib/liquid/profiler.rb:80:in `profile_node_render'
/Users/Jeiwan/Work/liquid/liquid/lib/liquid/profiler/hooks.rb:4:in `render_node_with_profiling'
/Users/Jeiwan/Work/liquid/liquid/lib/liquid/block_body.rb:72:in `block in render'
/Users/Jeiwan/Work/liquid/liquid/lib/liquid/block_body.rb:59:in `each'
/Users/Jeiwan/Work/liquid/liquid/lib/liquid/block_body.rb:59:in `render'
/Users/Jeiwan/Work/liquid/liquid/lib/liquid/template.rb:196:in `block in render'
/Users/Jeiwan/Work/liquid/liquid/lib/liquid/template.rb:230:in `with_profiling'
/Users/Jeiwan/Work/liquid/liquid/lib/liquid/template.rb:195:in `render'
/Users/Jeiwan/Work/liquid/liquid/lib/liquid/template.rb:208:in `render!'
/Users/Jeiwan/Work/liquid/liquid/test/integration/template_test.rb:297:in `block in test_undefined_filters_raise'
---------------

519 runs, 1277 assertions, 1 failures, 0 errors, 0 skips
rake aborted!
```

This happens only with `UndefinedFilter` exception because we don't actually raise it, but just handle it with `@context.handle_error` here: https://github.com/Shopify/liquid/pull/691/files#diff-cc7b2b233f09fa41b38a4a44e7c5bb3cR52 So this `exception_handler` executes not inside a `rescue`.
And we can't raise it because we don't want to interrupt the filter chain. So if we have filters like these: `{'foo' | undefinedfilter1 | upcase | undefinedfilter2}` we just want to end up with `FOO` and skip all undefined filters.